### PR TITLE
[WIP] LennardJonesFoce

### PR DESCRIPTION
**Pull request by ChayaSt**
This PR writes out `LennardJonesForce` if NBFIX exists and turns off `NonbondedForce` by zeroing the epsilon as discussed here: https://github.com/pandegroup/openmm/pull/1399#issuecomment-188586338 . 

We should probably add an option for future Charmm ffxml files (for new parameters not in the existing forcefield) to only write out L-J forces this way because most ions used in Charmm use NBFIX. Right now it only checks if there are NBFIXs in the current parameter set.

### CODE REVIEW

File: parmed/openmm/parameters.py [hunk 1]
```diff
@@ -466,6 +467,10 @@ def _write_omm_nonbonded(self, dest, skip_types):
                 sigma = 1.0
                 epsilon = 0.0
 
+            if self.nbfix_types:
+                # turn off L-J. Will use LennardJonesForce to use CostumNonbondedForce to compute L-J interactions
+                sigma = 1.0
+                epsilon = 0.0
```
Comment (swails) [inline parmed/openmm/parameters.py]: Perfect

**Comment by swails** [comment]
This looks great, @ChayaSt!

**Comment by swails** [comment]
Only thing is can you check on the Travis failure?  One of the tests is failing and I don't know why.

**Comment by ChayaSt** [comment]
My bad. I didn't realize you already had a ```toppar_water_ion.str``` file in the test files and added a newer version which has 11 NBFIXs rather than 6. I'll undo that commit.

**Comment by swails** [comment]
Is this still a WIP?  Let me know if/when you want me to merge it.

**Comment by ChayaSt** [comment]
I think we should wait until the generator is merged. I might still need to add changes here. Also, I want to add an option to write out ffxml files this way even if there's not NBFIX in the file. So if I want to add a str file for a newly parameterized molecule I can write out the ffxml file with the ```<LennardJonesForce>``` tag

**Comment by swails** [comment]
cc pandegroup/openmm#1453.  We're getting close to where we can revisit merging this in :)

**Comment by swails** [comment]
Any news on the status of this?

**Comment by swails** [comment]
Any news on this?  Do we still have plans to merge this in at some point?

**Comment by jchodera** [comment]
@ChayaSt : Can you tackle this on Monday?

**Comment by ChayaSt** [comment]
Yes, I'll take care of this on Monday.

**Comment by ChayaSt** [comment]
I added an option to write out ```<LennardJonesForce>``` and turn of ```sigma``` and ```epsilon``` for the charmm files that don't have ```NBFIX```. This ensures that future charmm files that are converted will be compatible with this design. You can merge this PR.

**Comment by jchodera** [comment]
Thanks!

File: parmed/openmm/parameters.py [hunk 2]
```diff
@@ -151,6 +151,12 @@ def write(self, dest, provenance=None, write_unused=True):
             templates remaining and parameters including those atom types will
             not be written. A ParameterWarning is issued if any such residues are
             found in a).
+        ljforce : bool
+            If True will turn off L-J and use the LennardJonesForce to use
+            CostumNonbondedForce to compute L-J interactions. Default is False. It
+            should be set to True when converting a charmm force field file that doesn't
+            have NBFIX in the file so that the ffxml conversion is compatible with the
+            main charmm36.xml file
```
Comment (swails) [inline parmed/openmm/parameters.py]: Add a note that when pair-specific L-J modifications are present (e.g., NBFIX in CHARMM), this behavior is always done and this flag is ignored.

**Comment by ChayaSt** [comment]
Fixed.

**Comment by swails** [comment]
Thanks!  It took awhile (opened in March), but I'll go ahead and merge this.

**Comment by swails** [comment]
Nice work!

**Pull request merged**